keras_core/layers/preprocessing/hashed_crossing_test.py

```diff
@@ -71,3 +78,90 @@ def test_tf_data_compatibility(self):
         for output in ds.take(1):
             output = output.numpy()
         self.assertAllClose(np.array([1, 4, 1, 1, 3]), output)
+
+    def test_upsupported_shape_input_fails(self):
+        with self.assertRaisesRegex(ValueError, "inputs should have shape"):
+            layers.HashedCrossing(num_bins=10)(
+                (tf.constant([[[1.0]]]), tf.constant([[[1.0]]]))
+            )
+
+    def test_cross_output_dtype(self):
+        if backend.backend() == "tensorflow" or backend.backend() == "numpy":
+            layer = layers.HashedCrossing(num_bins=2)
+            self.assertEqual(layer(([1], [1])).dtype, tf.int64)
+            layer = layers.HashedCrossing(num_bins=2, dtype=tf.int32)
+            self.assertEqual(layer(([1], [1])).dtype, tf.int32)
+            layer = layers.HashedCrossing(num_bins=2, output_mode="one_hot")
+            self.assertEqual(layer(([1], [1])).dtype, tf.float32)
+            layer = layers.HashedCrossing(
+                num_bins=2, output_mode="one_hot", dtype=tf.float64
+            )
+            self.assertEqual(layer(([1], [1])).dtype, tf.float64)
+
+        elif backend.backend() == "torch":
```

Comment:
Why this pattern? Can't we have the same tests for all backends? Just specify inputs via numpy arrays.

Reply:
You can use `backend.is_tensor()` to check that it's a tensor, and `backend.standardize_dtype(x.dtype)` to get a shared (string) dtype

Reply:
@fchollet Made these changes but there is a bug which is causing `jax` and `numpy` to fail.
For numpy the bug comes from following piece of code in `standardize_dtype`
```
    if dtype == "int":
        if config.backend() == "tensorflow":
            dtype = "int64"
        else:
            dtype = "int32"
```

As for Jax you check [here](https://github.com/google/jax/pull/4444/files) as to why it is truncating to `int32`. 

For numpy it is fixable need to change `standardize_dtype`  but for `Jax` either we can skip this test or let it remain as it is. What do you suggest ?